test/PowerShellEditorServices.Test.E2E/LSPTestsFixures.cs

```diff
@@ -2,58 +2,90 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json.Linq;
 using OmniSharp.Extensions.LanguageServer.Client;
-using OmniSharp.Extensions.LanguageServer.Client.Processes;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using Xunit.Abstractions;
 
 namespace PowerShellEditorServices.Test.E2E
 {
     public class LSPTestsFixture : TestsFixture
     {
         public override bool IsDebugAdapterTests => false;
 
-        public LanguageClient LanguageClient { get; private set; }
+        public ILanguageClient PsesLanguageClient { get; private set; }
         public List<Diagnostic> Diagnostics { get; set; }
 
+        public ITestOutputHelper Output { get; set; }
+
         public async override Task CustomInitializeAsync(
             ILoggerFactory factory,
-            StdioServerProcess process)
+            Stream inputStream,
+            Stream outputStream)
         {
-            LanguageClient = new LanguageClient(factory, process);
-
+            Diagnostics = new List<Diagnostic>();
             DirectoryInfo testdir =
                 Directory.CreateDirectory(Path.Combine(s_binDir, Path.GetRandomFileName()));
 
-            await LanguageClient.Initialize(testdir.FullName);
+            PsesLanguageClient = LanguageClient.PreInit(options =>
+            {
+                options
+                    .WithInput(inputStream)
+                    .WithOutput(outputStream)
+                    .WithRootUri(DocumentUri.FromFileSystemPath(testdir.FullName))
+                    .OnPublishDiagnostics(diagnosticParams =>
+                    {
+                        Diagnostics.AddRange(diagnosticParams.Diagnostics.Where(d => d != null));
+                    })
+                    .OnLogMessage(logMessageParams =>
+                    {
+                        Output?.WriteLine($"{logMessageParams.Type.ToString()}: {logMessageParams.Message}");
+                    });
+
+                // Enable all capabilities this this is for testing.
+                var capabilities = typeof(ICapability).Assembly.GetExportedTypes()
+                    .Where(z => typeof(ICapability).IsAssignableFrom(z))
+                    .Where(z => z.IsClass && !z.IsAbstract);
+                foreach (var item in capabilities)
```

Comment:
What are the `var`s here? Possible to change them to explicit types?

Reply:
updated. Also added a comment that this code is temporary.